src/fhirtypes/StructureDefinition.ts

```diff
@@ -286,6 +287,10 @@ export class StructureDefinition {
     if (path.startsWith('snapshot') || path.startsWith('differential')) {
       throw new InvalidElementAccessError(path);
     }
+    const parentName = this.type || 'Resource';
+    if (path == 'type' && value != parentName) {
```

Comment:
Using `==` does work fine, but I think it is more conventional in this case to use `===` and `!==`. The triple equals checks for equality of value and type. The double equals can still return true even when things types don't match, for example, something like `'1' == 1` is true, but `'1' === 1` is false. In this case the distinction would probably never matter, but it's still probably better to be safe and use `===`.